### DISCUSSION TITLE
How is ROCm support for RX 7700 XT? I'm thinking of buying it for Stable Diffusion and other similar projects, like RVC, etc.

### DISCUSSION THREAD

**Discussion by linuxtroubleshooting** [ROCm/ROCm#2769] [Q&A]
How is ROCm support for RX 7700 XT? I'm thinking of buying it for Stable Diffusion and other similar projects, like RVC, etc.

**Comment by ChuloIva** [comment]
Also interested in this, been searching everywhere couldn't find anything. Looking to run local LLMs

**Reply by james-banks** [reply]
It's not too bad, but you have to be prepared to use Linux and play around in the terminal to get things set up properly. Like the 7700, my 7800 XT isn't "officially" supported by ROCm but it works.

The other alternatives if you want to use windows include ONNX and DirectML, but personally I've been using ZLUDA (a CUDA translation layer that uses AMD's HIP SKD) on windows for the last week and it's been awesome. With SD1.5 at 512x512 on my RX 7800 XT I get about 8 it/s on SD.Next and closer to 20 it/s when using the command line version.

**Reply by yuu-fur** [reply]
hey @james-banks can you link to something that explains more on how to make the 7700xt work with ROCm

**Reply by james-banks** [reply]
@Yuu-to-kun Sure, this guide on the AMD website should give you everything you need: https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/install-radeon.html

This gives guidance for the ROCm specific PyTorch version install: https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/install-pytorch.html

Only other note is that you need to update the environment variable  `export HSA_OVERRIDE_GFX_VERSION=11.0.0` to make it think you have a compatible (ie RX7900 XT) card. Then you can just install SD from GitHub and it should work.

There's a slightly old but still relevant full guide here, just update the numbers to the latest versions (i.e use the Pytorch install command given on the second link above instead, otherwise you'll install an old version):  https://www.reddit.com/r/StableDiffusion/comments/10zfnlj/novice_guide_how_to_fully_setup_linux_to_run/

**Reply by ChuloIva** [reply]
I've found this

https://www.reddit.com/r/LocalLLaMA/comments/170tghx/guide_installing_rocmhip_for_llamacpp_on_linux/?rdt=38080&onetap_auto=true&one_tap=true

did a fresh install of Ubuntu 22.04.4LTS (was on 23.10)

and it worked for me. Didn't even had to do some workarounds that were mentioned, like the error for: 
sudo apt install rocm-dev

**Reply by yuu-fur** [reply]
Very much appreciated @james-banks, the export did the trick as well as the amd instructions for pytorch. Since i use arch i didn't have to install the amdgpu-install since we have an arch repository for that. For anyone else that happens on this comment, i use arch and have an 7700xt and all i had to do was follow the official instructions from the stable-diffusion-web-ui amd installation and instead of installing pytorch and pytorch vision from the repos mentioned just use the ones that amd officially provides. And before any installation make sure to do the export mentioned above.

**Comment by pereira-rafael** [comment]
Im already have one rx 7700xt but i havert sucess to use ROCm =/

**Reply by Pranavhc** [reply]
any update? I just got myself a RX 7700XT and I've run into the same problem

**Reply by pereira-rafael** [reply]
I managed to use ROCm on Ubuntu after changing a variable with the following command:

export HSA_OVERRIDE_GFX_VERSION=11.0.0

This way, for ROCm, the RX 7700 XT is recognized as a 7800 XT.

**Reply by denisgolius** [reply]
I tried a few times yesterday, but pytorch said that there is no cuda, maybe because it works in python venv?
FYI: I used https://phazertech.com/tutorials/rocm.html to install https://repo.radeon.com/amdgpu-install/6.4/ubuntu/noble/ .

**Reply by pereira-rafael** [reply]
In my case, I used a Docker container featuring the ROCm-adapted PyTorch made available by AMD. Within the container, I executed the aforementioned command to "trick" ROCm, which then allowed me to use ROCm normally.

Follow link with amd tutorial 

https://rocm.docs.amd.com/projects/install-on-linux/en/latest/index.html

**Reply by denisgolius** [reply]
Thanks! I'll try

> In my case, I used a Docker container featuring the ROCm-adapted PyTorch made available by AMD. Within the container, I executed the aforementioned command to "trick" ROCm, which then allowed me to use ROCm normally.
> 
> Follow link with amd tutorial
> 
> https://rocm.docs.amd.com/projects/install-on-linux/en/latest/index.html

Could you share your Dockerfile?

**Reply by pereira-rafael** [reply]
On link below there guide step how to download and use the docker image

https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/3rd-party/pytorch-install.html#using-docker-with-pytorch-pre-installed

**Comment by harkgill-amd** [comment]
ROCm 10 includes support for the 7700XT/gfx1102, see https://rocm.docs.amd.com/en/latest/install/rocm.html. Workloads such as Stable Diffusion, llama.cpp.etc will work OOB but if you do run into any issues, feel free to post a GitHub issue and we'll take a look from there.

**Reply by denisgolius** [reply]
better later then never... but I'v switched to rtx 5070ti last June before AI boom
